assets/content/cookbook/Data structures/ring-array.md

```diff
@@ -0,0 +1,140 @@
+[tags]: / "collections"
+
+# A fixed ring array
+
+A fixed ring array are especially useful when you need a hard upper bound for how much data can be in the queue.
+
+```haxe
+#if !js @:generic #end
```

Comment:
I don't think this is needed? Same code seems to be generated with and without it.

Reply:
Because I feel like `@:generic` can get better performance on static platforms, So I disabled it only in js

Reply:
Sure, but my point is that you don't think you _need_ to disable it on JS, as it doesn't seem to do actually do anything there.

Reply:
ok. I have removed conditional compilation to make it look simpler